specification/cognitiveservices/OpenAI.Inference/models/completions/chat_messages.tsp

```diff
@@ -0,0 +1,202 @@
+model ChatRequestMessage {
+  @doc("The chat role associated with this message.")
+  role: ChatRole;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("""
+A request chat message containing system instructions that influence how the model will generate a chat completions
+response.
+""")
+model ChatRequestSystemMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'system' for system messages.")
+  role: ChatRole.system;
+
+  @doc("The contents of the system message.")
+  content: string;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A request chat message representing user input to the assistant.")
+model ChatRequestUserMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'user' for user messages.")
+  role: ChatRole.user;
+
+  @doc("The contents of the user message, with available input types varying by selected model.")
+  content: ChatMessageContent;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A request chat message representing response or action from the assistant.")
+model ChatRequestAssistantMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'assistant' for assistant messages.")
+  role: ChatRole.assistant;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly nullable in mirrored API"
+  @doc("The content of the message.")
+  content: string | null;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+
+  @projectedName("json", "tool_calls")
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @doc("""
+  The tool calls that must be resolved and have their outputs appended to subsequent input messages for the chat
+  completions request to resolve as configured.
+  """)
+  toolCalls?: ChatCompletionsToolCall[];
+
+  @added(ServiceApiVersions.v2023_07_01_Preview)
+  @removed(ServiceApiVersions.v2023_12_01_Preview)
+  @projectedName("json", "function_call")
+  @doc("""
+  The function call that must be resolved and have its output appended to subsequent input messages for the chat
+  completions request to resolve as configured.
+  """)
+  functionCall?: FunctionCall;
+}
+
+@added(ServiceApiVersions.v2023_12_01_Preview)
+@doc("A request chat message representing requested output from a configured tool.")
+model ChatRequestToolMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'tool' for tool messages.")
+  role: ChatRole.tool;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly nullable in mirrored API"
+  @doc("The content of the message.")
+  content: string | null;
+
+  @projectedName("json", "tool_call_id")
+  @doc("The ID of the tool call resolved by the provided content.")
+  toolCallId: string;
```

Comment:
This [line](https://github.com/Azure/azure-rest-api-specs/blob/b456e86a719c459fbd0a5b096b37a2211a519e84/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-12-01-preview/inference.yaml#L1086) makes me think that `toolCallId` might also be nullable.

Reply:
[OpenAI spec](https://github.com/openai/openai-openapi/blob/a7a0b63d1fac3acac9a223d57da1219c8a97c01e/openapi.yaml#L5704C6-L5704C6) has it required, non-nullable.